pinot-core/src/test/java/org/apache/pinot/core/query/reduce/StreamingReduceServiceTest.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.reduce;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Iterator;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+import org.apache.pinot.common.proto.Server;
+import org.apache.pinot.core.transport.ServerRoutingInstance;
+import org.apache.pinot.spi.config.table.TableType;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.testng.annotations.Test;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertTrue;
+
+
+public class StreamingReduceServiceTest {
+
+  @Test
+  public void testThreadExceptionTransfer() {
+    // simulate a thread exception in gRPC call and verify that the thread can transfer the exception
+    Iterator<Server.ServerResponse> mockedResponse = (Iterator<Server.ServerResponse>) mock(Iterator.class);
+    when(mockedResponse.hasNext()).thenReturn(true);
+    final String exceptionMessage = "Some exception";
+    final RuntimeException innerException = new RuntimeException(exceptionMessage);
+    when(mockedResponse.next()).thenThrow(innerException);
+    final ExecutorService threadPoolService = Executors.newFixedThreadPool(1);
+    final ServerRoutingInstance routingInstance =
```

Comment:
It's not enforced but declaring variables final is against the project style, these are all effectively final so can be referenced in a nested scope anyway.

Reply:
Thanks. Did not know we prefer not using final on local variables. Fixed.